modules/account-lib/src/coin/dot/transferBuilder.ts

```diff
@@ -1,26 +1,197 @@
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { decode, methods } from '@substrate/txwrapper-polkadot';
+import BigNumber from 'bignumber.js';
+import { InvalidTransactionError } from '../baseCoin/errors';
 import { TransactionBuilder } from './transactionBuilder';
 import { Transaction } from './transaction';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TransactionType } from '../baseCoin';
+import { MethodNames, ProxyArgs, proxyType, TransferArgs } from './iface';
+import { ProxyTransactionSchema, TransferTransactionSchema } from './txnSchema';
+import utils from './utils';
+import { BaseAddress } from '../baseCoin/iface';
 
 export class TransferBuilder extends TransactionBuilder {
+  protected _amount: string;
+  protected _to: string;
+  protected _real: string;
+  protected _forceProxyType: proxyType;
+
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
+  /**
+   *
+   * Dispatch the given call from an account that the sender is authorised for through add_proxy.
+   *
+   * @returns {UnsignedTransaction} an unsigned Dot transaction
+   *
+   * @see https://polkadot.js.org/docs/substrate/extrinsics/#proxy
+   */
+  protected buildTransaction(): UnsignedTransaction {
+    const baseTxInfo = this.createBaseTxInfo();
+    const transferTx = methods.balances.transferKeepAlive(
+      {
+        value: this._amount,
+        dest: this._to,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+    if (!this._real) {
+      return transferTx;
+    }
+    return methods.proxy.proxy(
+      {
+        real: this._real,
+        forceProxyType: this._forceProxyType,
+        call: transferTx.method,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.Send;
+  }
+
+  /**
+   *
+   * The amount for transfer transaction.
+   *
+   * @param {string} amount
+   * @returns {TransferBuilder} This transfer builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-protocol-info
+   */
+  amount(amount: string): this {
+    this.validateValue(new BigNumber(amount));
+    this._amount = amount;
+    return this;
+  }
+
+  /**
+   *
+   * The destination address for transfer transaction.
+   *
+   * @param {string} dest
+   * @returns {TransferBuilder} This transfer builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-protocol-info
+   */
+  to({ address }: BaseAddress): this {
+    this.validateAddress({ address });
+    this._to = address;
+    return this;
+  }
+
+  /**
+   *
+   * The real address of the original tx
+   *
+   * @param {BaseAddress} real
+   * @returns {TransferBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#why-use-a-proxy
+   */
+  real(real: BaseAddress): this {
```

Comment:
We need a better name for this. Add it to the list of follow-up changes we need to make. This sounds like the `owner`